.github/workflows/build.yml

```diff
@@ -86,52 +84,36 @@ jobs:
           restore-keys: |
             depends-sources-
 
+      - name: Determine params
+        id: det-params
+        run: |
+          dep_name="${{ matrix.depends_name }}"
+          dep_opts="${{ matrix.dep_opts }}"
+          dep_hash="$(echo -n ${dep_opts} | sha256sum | head -c 64)"
+          echo "\"${dep_name}\" has DEP_OPTS \"${dep_opts}\" with hash \"${dep_hash}\""
+          echo "dep_hash=${dep_hash}" >> $GITHUB_OUTPUT
+
       - name: Cache depends
         uses: actions/cache@v4
         with:
           path: |
             depends/built
             depends/${{ matrix.host }}
-          key: ${{ runner.os }}-depends-${{ matrix.build_target }}-${{ hashFiles('depends/packages/*') }}
+          key: ${{ runner.os }}-depends-${{ matrix.depends_name }}-${{ hashFiles('depends/packages/*') }}-${{ steps.det-params.outputs.dep_hash }}
           restore-keys: |
-            ${{ runner.os }}-depends-${{ matrix.build_target }}-${{ hashFiles('depends/packages/*') }}
-            ${{ runner.os }}-depends-${{ matrix.build_target }}
+            ${{ runner.os }}-depends-${{ matrix.depends_name }}-${{ hashFiles('depends/packages/*') }}-
+            ${{ runner.os }}-depends-${{ matrix.depends_name }}-
 
       - name: Build depends
-        run: make -j$(nproc) -C depends HOST=${{ matrix.host }}
+        run: env HOST=${{ matrix.host }} ${{ matrix.dep_opts }} make -j$(nproc) -C depends
 
   build:
-    name: Build
+    name: ${{ matrix.build_target }}
     needs: [build-image, build-depends]
     runs-on: ubuntu-22.04
     strategy:
       fail-fast: false
-      matrix:
-        include:
-          - build_target: arm-linux
-            host: arm-linux-gnueabihf
-            depends_on: arm-linux
-          - build_target: linux64
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_tsan
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_ubsan
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_fuzz
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_cxx20
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_sqlite
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
-          - build_target: linux64_nowallet
-            host: x86_64-pc-linux-gnu
-            depends_on: linux64
+      matrix: ${{ fromJson(needs.build-image.outputs.src-matrix) }}
```

Comment:
I fear this overcomplicates build.yml and will make it harder to improve / debug / logic on

Reply:
Outputs have to be generated in the previous job in order to be used in the current job and creating a new job just to read the matrix seems a bit wasteful, so the build image step is being used to also read, generate and export as output, the matrices needed for the next two jobs.

We can probably try reusable workflows ([source](https://docs.github.com/en/actions/sharing-automations/reusing-workflows)) in the future, it'll be more GitLab-like but would invite duplication and a different kind of convoluted syntax. Depends on how reliable this setup proves to be.